df_chat/tests/test_message_endpoint.py

```diff
@@ -26,6 +31,110 @@ def test_message_creation_success(self):
+            # Assert the response body contains the keyword.
+            self.assertIn(
+                body.lower(), messages_list_response.json()[0]["body"].lower()
+            )
+
+        # * 1. Keyword search - not ok cases
+        for body in [
+            "somethingrandom",
+            "messages",
+            "124",
+        ]:
+            messages_list_response = await self.async_client.get(
+                messages_list_api_url,
+                {"body": body},
+            )
+            # Assert the correct HTTP status code for response.
+            self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+
+        # * 2. username search - ok case
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"username": user.username},
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+        # Assert the message `room_user_id` matches the `id` field of the `RoomUser` object
+        #   corresponding to the user.
+        room_user_obj = RoomUser.objects.get(user__id=user.id)
+        self.assertEqual(
+            messages_list_response.json()[0]["room_user_id"], room_user_obj.id
+        )
+
+        # * 2. username search - not ok case
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"username": "invalid_username"},
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+
+        now = datetime.now()
+        before = now - timedelta(hours=1)
+        datetime_format = "%Y-%m-%dT%H:%M:%S.%f"
+
+        # * datetime range search - range start
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"created_gte": str(before)},
+        )
+        message_created = datetime.strptime(
+            messages_list_response.json()[0]["created"], datetime_format
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+        # Assert message `created` datetime is larger than `before` datetime.
+        self.assertTrue(message_created >= before)
+
+        # * datetime range search - range end
+        messages_list_response = await self.async_client.get(
+            messages_list_api_url,
+            {"created_lte": str(now)},
+        )
+        message_created = datetime.strptime(
+            messages_list_response.json()[0]["created"], datetime_format
+        )
+        # Assert the correct HTTP status code for response.
+        self.assertEqual(messages_list_response.status_code, http.HTTPStatus.OK)
+        # Assert message `created` datetime is smaller than `now` datetime.
+        self.assertTrue(message_created <= now)
```

Comment:
Please add a test where  the filter is "{"created_lte": str(before)}", to ensure that the endpoint does not return any messages that are created after "before" time.

Reply:
Done!